ui/nuxeo-document-comments/nuxeo-document-comment-thread.js

```diff
@@ -0,0 +1,293 @@
+      return 'nuxeo-document-comment-thread';
+    }
+
+    static get properties() {
+      return {
+        uid: {
+          type: String,
+          observer: '_refresh',
+        },
+
+        comments: {
+          type: Array,
+          value() {
+            return [];
+          },
+        },
+
+        level: {
+          type: Number,
+          value: 1,
+        },
+
+        pageSize: {
+          type: Number,
+          readOnly: true,
+          value: 10,
+        },
+
+        allCommentsLoaded: {
+          type: Boolean,
+          readOnly: true,
+          value: false,
+        },
+
+        total: {
+          type: Number,
+          readOnly: true,
+          value: 0,
+        },
+      };
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+      this.addEventListener('delete-comment', this._handleDeleteEvent);
+      this.addEventListener('edit-comment', this._handleEditEvent);
+      this.addEventListener('comments-changed', this._handleCommentsChange);
+    }
+
+    disconnectedCallback() {
+      this.removeEventListener('delete-comment', this._handleDeleteEvent);
+      this.removeEventListener('edit-comment', this._handleEditEvent);
+      this.removeEventListener('comments-changed', this._handleCommentsChange);
+      super.disconnectedCallback();
+    }
+
+    focusInput() {
+      this.$$('#inputContainer').focus();
+    }
+
+    _checkForEnter(e) {
+      if (e.keyCode === 13 && e.ctrlKey) {
+        if (!this._isBlank(this.text)) {
+          this._submitComment();
+        }
+      }
+    }
+
+    _clearInput() {
+      this.text = '';
+    }
+
+    _clearRequest() {
+      this.$.commentRequest.data = {};
+      this.$.commentRequest.headers = {};
+      this.$.commentRequest.params = {};
+    }
+
+    _fetchComments(loadAll) {
+      this._clearRequest();
+      this.$.commentRequest.params = {
+        pageSize: loadAll ? 0 : this.pageSize,
+        currentPageIndex: 0,
+      };
+      this.$.commentRequest.headers = {
+        'X-NXfetch.comment': 'repliesSummary',
+      };
+      this.$.commentRequest
+        .get()
+        .then((response) => {
+          /* Reconciliation of local and server comments */
+          const olderComment = this.comments.length > 0 ? this.comments[0] : null;
+          const newComments = response.entries;
+          while (
+            newComments.length > 0 &&
+            !!olderComment &&
+            (newComments[0].creationDate > olderComment.creationDate || newComments[0].id === olderComment.id)
+          ) {
+            newComments.shift();
+          }
+          response.entries.forEach((entry) => {
+            this.unshift('comments', entry);
+          });
+          this._setTotal(response.totalSize);
+          this._setAllCommentsLoaded(!!loadAll);
+        })
+        .catch((error) => {
+          if (error.status === 404) {
+            this.fire('notify', { message: this._computeTextLabel(this.level, 'notFound') });
```

Comment:
I think we're missing the `@event notify`  JSDoc which should make the event visible in the storybook actions panel